src/Ocelot/DownstreamUrlCreator/Middleware/DownstreamUrlCreatorMiddleware.cs

```diff
@@ -90,6 +92,33 @@ public async Task Invoke(HttpContext httpContext)
             await _next.Invoke(httpContext);
         }
 
+        private static string MergeQueryStringsWithoutDuplicateValues(string queryString, string newQueryString)
+        {
+            var queries = HttpUtility.ParseQueryString(queryString);
+            var newQueries = HttpUtility.ParseQueryString(newQueryString);
+
+            var dict = new Dictionary<string, string>();
+            foreach (var key in newQueries.AllKeys)
+            {
+                if (!string.IsNullOrEmpty(key))
+                {
+                    dict.Add(key, newQueries[key]);
+                }
+            }
+
+            foreach (var key in queries.AllKeys)
+            {
+                if (string.IsNullOrEmpty(key) || dict.ContainsValue(queries[key]))
+                {
+                    continue;
+                }
+
+                dict.Add(key, queries[key]);

```

Comment:
The main concern about usage of `Add` method is that this method generates exception when duplicate key has detected. Why not to use `TryAdd` instead of `Add`?
```suggestion
                dict.TryAdd(key, queries[key]);
```

Reply:
Fixed in https://github.com/ThreeMammals/Ocelot/pull/1182/commits/fef94b63b48cea1bbdd36180001e4e44d5586905